File: code/6._Run-Length_Encoding(RLE)/Python/programming.py/RLE.py
def run_length_encode(input_string):
    encoded = ""
    count = 1
    for i in range(1, len(input_string)):
        if input_string[i] == input_string[i - 1]:
            count += 1
        else:
            encoded += input_string[i - 1] + str(count)
            count = 1
    encoded += input_string[-1] + str(count)
    return encoded


def run_length_decode(encoded_string):
    decoded = ""
    i = 0
    while i < len(encoded_string):
        char = encoded_string[i]
        count = int(encoded_string[i + 1])
        decoded += char * count
        i += 2
    return decoded


# Example usage:
input_str = "WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB"
encoded_str = run_length_encode(input_str)
print("Encoded:", encoded_str)
decoded_str = run_length_decode(encoded_str)
print("Decoded:", decoded_str)

File: code/6._Run-Length_Encoding(RLE)/PHP/programming.php/RLE.php
function runLengthEncode($input) {
$encoded = '';
$count = 1;
for ($i = 1; $i < strlen($input); $i++) { if ($input[$i]==$input[$i - 1]) { $count++; } else { $encoded .=$input[$i - 1]
    . $count; $count=1; } } $encoded .=$input[strlen($input) - 1] . $count; return $encoded; } function
    runLengthDecode($encoded) { $decoded='' ; $len=strlen($encoded); for ($i=0; $i < $len; $i +=2) {
    $character=$encoded[$i]; $count=(int)$encoded[$i + 1]; $decoded .=str_repeat($character, $count); } return $decoded;
    } $input="WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB" ; $encoded=runLengthEncode($input);
    echo "Encoded: " . $encoded . "\n" ; $decoded=runLengthDecode($encoded); echo "Decoded: " . $decoded . "\n" ;